### PR TITLE
update installation in setup.py and installlation documentation

### LINKED ISSUE #740: authorship of Solaar in setup.py
Daniel Pavel is listed as the sole author of Solaar in setup.py 

As far as I can tell, this puts him and his email in several repositories, such as PyPI https://pypi.org/project/solaar/

Who should be put there?

### LINKED ISSUE #741: manual install instructions incorrect with respect to plugdev group
The manual install instructions refer to  the plugdev group, but Solaar no longer uses that method of obtaining R/W access to the receiver device.

### LINKED ISSUE #742: setup.py autostart file determination not effective
setup.py looks at sys.argv to determine which arguments were used but that no longer works.  

This is only used to determine where to put the autostart file.

I'm going to take out the code that is not working.   This will not change how setup.py works but will remove a red herring.

### PR DESCRIPTION
**Pull request by pfps**
Remove unusable code in setup.py
Install udev rule from setup.py
Update installation documentation
Remove incorrect rules/install.sh file

Fixes #742 
Fixes #741
Fixes #740

### CODE REVIEW

**Comment by FFY00** [comment]
Did you test the instalation changes? Make usre it works both when installed via `setup.py` and pip.

**Comment by pfps** [comment]
So far only pip.   I'll run them through the other way in a moment.
I did do both --user and global.

**Comment by pfps** [comment]
`python setup.py --user` doesn't work because of the absolute locations for the autostart file (and now also the udev rule).

As far as I can tell, `sudo python setup.py install` hasn't worked for a  little while.
The installation looks OK, but starting solaar crashes when trying to initialize the main window because the icon file names are not set up correctly for some reason.

I'll try to find out where this was introduced.

**Comment by pfps** [comment]
I found the commit that broke `sudo python setup.py install`
It is 
```
commit 6ba8c6c3fc5d7727f1fdba782014506cd9253127
Author: drupal-daffie <drupal-daffie@users.noreply.github.com>
Date:   Tue Dec 24 13:16:35 2019 +0100

    setup: use setuptools if available
```

My guess is that when the setuptools setup is run outside of pip Solaar cannot find the icon files.

To see this breakage you have to be careful to remove all the icon files that Solaar installs (and maybe not have Solaar icon files in a theme).


@FFY00  My view is that installation should only be done with pip.  If you agree I'll put that in the installation instructions.

**Comment by FFY00** [comment]
Packagers need to install with `setup.py`, so we should definitely fix this before a new release.

**Comment by pfps** [comment]
Given that the Ubuntu git PPA package has been working part of the time since that stage, it appears that there is enough of `python setup.py ...` working for packaging to distributions.  But it would be nice to have some better confirmation of this.
But can't distributions just use pip to generate the files needed for a package?

Is there any reasonable use for `python setup.py install --user`?

**Comment by pfps** [comment]
`python3 setup.py sdist bdist_wheel` definitely works as I've been using it to create stuff for PyPI.

**Comment by pfps** [comment]
The problem is that with setuptools `sudo python3 setup.py install` installs the icon files under
/usr/local/lib/python3.7/site-packages/solaar-1.0.2rc1-py3.7.egg/shared/solaar instead of under /usr/local/share/solaar.   This is a feature, probably so as not to override files that have more official status.

**Comment by pfps** [comment]
The icon problem is fixed by PR #745

**Comment by pfps** [comment]
Should I add (back) in code to handle `python3 setup.py install --user`?
Users should use `pip install --user .` not this method.

File: setup.py [hunk 1]
```diff
@@ -56,11 +39,10 @@ def _data_files():
 battery status.
 '''.strip(),
 		author='Daniel Pavel',
-		author_email='daniel.pavel@gmail.com',
 		license='GPLv2',
 		url='http://pwr-solaar.github.io/Solaar/',
 		classifiers=[
-			'Development Status :: 4 - Beta',
+			'Development Status :: 5 - Production/Stable',
```
Comment (FFY00) [inline setup.py:45]: Not sure about this.

Comment (pfps) [reply setup.py:45]: Me neither, but version 1 generally implies production.

**Comment by pfps** [comment]
I just checked all the build commands.  Everything useful works
```
pip install . --user
sudo pip install .
sudo python3 setup.py install
python3 setup.py sdist bdist-wheel
````

`python3 setup.py install --user` does not, but not one should ever do that!  Nor should anyone do `sudo python3 setup.py install`

File: setup.py (hunk 1 above)
Comment (pfps) [reply setup.py:45]: But I'll change it back and we can revisit at the next release.

**Comment by pfps** [comment]
OK, should be ready to go.

**Pull request merged**